src/main/java/org/ecocean/Shepherd.java

```diff
@@ -2002,6 +2002,44 @@ public int getNumTaxonomies() {
         return (Util.count(taxis));
     }
 
+    public List<List<String>> getAllTaxonomyCommonNames() {return getAllTaxonomyCommonNames(false);
+    }
+
+    // forceSpaces will turn `Foo bar_bar` into `Foo bar bar` - use with caution!
+    public List<List<String>> getAllTaxonomyCommonNames(boolean forceSpaces) {
+        Set<String> allSciNames = new HashSet<String>();
+        Set<String> allComNames = new HashSet<String>();
+
```

Comment:
one more fix that slipped by me on first review! if you can change these to `LinkedHashSet`, it will solve a problem where the two lists (below) which are created from these HashSets **get out of order**.  a regular HashSet has an arbitrary order, but LinkedHashSet keeps the order they were inserted; which turns out to work perfectly as a fix here -- so the scientific names match up with their correct common names.